### ISSUE TITLE
Cost to run SWE-Bench Lite

### ISSUE THREAD

**Issue by JensRoland** [aorwall/moatless-tree-search#11]
Could you include the approximate cost to run the SWE-Bench Lite benchmark? Ideally as a histogram of cost-per-task, but a total cost for the particular LLM used would be fine too.

As coding agent frameworks approach the threshold for actual viability, operating ROI starts to be a factor, especially given how some frameworks/providers seem to be spending 10-20x more money to eke out a few more percent on SWE-Bench. I feel like Moatless is one of the few high performers actively pursuing cost effectiveness, so I would feature these numbers prominently.

**Comment by aorwall** [comment]
That’s a very good idea. I currently write out the cost per instance in my comment for each eval run, but I guess it’s easy to miss. Do you have any suggestions for a better way to present this more clearly?

**Comment by JensRoland** [comment]
In two ways:

First, as a single number in the README (cost to run the SWE-Bench Lite set)

Second, because a single number can be misleading, as a histogram with cost 'buckets' along the x-axis (perhaps $0.05 each?) and the number of tasks in SWE-Bench Lite that fell inside that cost bucket.

This would communicate a lot:

- The (rough) average can be inferred trivially
- The variance too
- The number of tasks which hit the max limit (likely due to a logical loop)
- An expected range for 'simple' vs. 'hard' tasks in practice

These are all relevant properties that are clearly communicated (even to laymen) with a histogram plot.

**Comment by JensRoland** [comment]
Also, if you want to automate it, the <https://github.com/AgentOps-AI/tokencost> package seems like a good source (if not to use directly, then at least to grab the latest price list)

**Comment by aorwall** [comment]
Looks like something that could be added to a hosted streamlit app and then a short summary in the readme file.

**Comment by aorwall** [comment]
Seems like $5000 dollars was spent to achieve the new SOTA on SWE-bench Verified. It proves your point I’d say. https://x.com/skcd42/status/1867586492253581403

**Comment by JensRoland** [comment]
Yes… when I read that AlphaCode may go through >100k attempts at solving a single task, it became clear that we shouldn’t be so quick to celebrate SotA performance without considering the cost dimension.

Now that X post mentions the bitter lesson, but I’m not convinced that the NN bitter lesson applies to agentic system inference. I think what they did is more akin to brute forcing password files using a mutating password dictionary. Sure, it works, but [PCFG and LSTM approaches](https://wiki.ubc.ca/Course:CPSC522/Password_cracking_using_PCFGs_and_Neural_Networks) work vastly better. In my mind, optimal performance in this space should look more like PCFG and less like brute force.

**Comment by aorwall** [comment]
I've set up the page https://experiments.moatless.ai now, where there's a bit more information about the cost of running things.

**Comment by JensRoland** [comment]
Love it! With that, I think this ticket can be closed... Though it would be nice if those tables had a 'sort by column' feature.

And also if the tables included Moatless-Tree-Search :-)
